src/Features/LanguageServer/Protocol/RoslynLanguageServer.cs

```diff
@@ -82,6 +81,13 @@ protected override IRequestExecutionQueue<RequestContext> ConstructRequestExecut
             AddBaseService<IMethodHandler>(new InitializedHandler());
             AddBaseService<IOnInitialized>(this);
 
+            // In all VS cases, we already have a misc workspace.  Specifically
+            // Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.MiscellaneousFilesWorkspace.  In
+            // those cases, we do not need to add an additional workspace to manage new files we hear about.  So only
+            // add the LspMiscellaneousFilesWorkspace for hosts that have not already brought their own.
+            if (serverKind == WellKnownLspServerKinds.CSharpVisualBasicLspServer)
```

Comment:
slightly worried about brittleness here.  @dibarbet is this ok for WellKnownLspServerKinds.Razor?

Reply:
I believe it should be fine - the `CSharpVisualBasicLspServer` should be the only server that is used outside of VS.